features/login/impl/src/main/kotlin/io/element/android/features/login/impl/changeserver/ChangeServerState.kt

```diff
@@ -23,5 +23,5 @@ data class ChangeServerState(
     val changeServerAction: Async<Unit>,
     val eventSink: (ChangeServerEvents) -> Unit,
 ) {
-    val submitEnabled = homeserver.isNotEmpty() && changeServerAction is Async.Uninitialized
+    val submitEnabled: Boolean get() = homeserver.isNotEmpty() && (changeServerAction is Async.Uninitialized || changeServerAction is Async.Loading)
```

Comment:
Should be enabled when `changeServerAction is Async.Failure` too. So in fact I think the test can be removed, and it should be simply:

```suggestion
    val submitEnabled: Boolean get() = homeserver.isNotEmpty()
```

WDYT?

Reply:
Similar to the comment below, in this case the retry is blocked and the error is displayed until the user changes the info in the text field, either adding or removing some text. As soon as that happens, the error is cleared and the user can retry.